OMCompiler/Compiler/NBackEnd/Modules/1_Main/NBInitialization.mo

```diff
@@ -75,9 +77,16 @@ public
           qual.varData := varData;
           qual.eqData := eqData;
       then qual;
+
+      else algorithm
+        Error.addMessage(Error.INTERNAL_ERROR,{getInstanceName() + " failed!"});
+      then fail();
     end match;
 
+    // Modules
     bdae := Partitioning.main(bdae, NBSystem.SystemType.INIT);
+    bdae := Causalize.main(bdae, NBSystem.SystemType.INIT);
+    bdae := Tearing.main(bdae, NBSystem.SystemType.INIT);
```

Comment:
totally! 

> Just a thought, something I saw in SimCode. (Asserting all modules will have the same interface) what do you think?

Yes totally! That is how we want to do it in the future. The current version in `BackendDAEUtil.mo` is pretty good i think... there we have something like that but the list it traverses depends on the pre set optimization modules. I am not quite sure if we want to stick with string literals, but i don't really see a reason why not since the user has to put them in as string literals for command line options anyway.